client/ayon_core/hosts/maya/plugins/load/load_maya_usd_add_maya_reference.py

```diff
@@ -0,0 +1,155 @@
+# -*- coding: utf-8 -*-
+import contextlib
+
+from ayon_core.pipeline import load
+from ayon_core.hosts.maya.api.usdlib import (
+    containerise_prim,
+    iter_ufe_usd_selection
+)
+
+from maya import cmds
+import mayaUsd
+
+
+@contextlib.contextmanager
+def no_edit_mode(prim, restore_after=True):
+    """Ensure MayaReference prim is not in edit mode during context"""
+    pulled_node = mayaUsd.lib.PrimUpdaterManager.readPullInformation(prim)
+    ufe_path = None
+    try:
+        # remove edit state if pulled
+        if pulled_node:
+            import mayaUsdUtils
+            assert mayaUsdUtils.isPulledMayaReference(pulled_node)
+            cmds.mayaUsdDiscardEdits(pulled_node)
+
+            # Discarding the edits directly selects the prim
+            # so we can get the UFE path from selection
+            ufe_path = cmds.ls(selection=True, ufeObjects=True, long=True)[0]
+
+        yield prim, ufe_path, pulled_node
+    finally:
+        if restore_after and pulled_node and ufe_path:
+            cmds.mayaUsdEditAsMaya(ufe_path)
+
+
+class MayaUsdProxyAddMayaReferenceLoader(load.LoaderPlugin):
+    """Read USD data in a Maya USD Proxy
+
+    TODO: It'd be much easier if this loader would be capable of returning the
+        available containers in the scene based on the AYON URLs inside a USD
+        stage. That way we could potentially avoid the need for custom metadata
+        keys, stay closer to USD native data and rely solely on the
+        AYON:asset=blue,subset=modelMain,version=1 url
+
+    """
+
+    product_types = {"*"}
+    representations = ["*"]
+    extensions = ["ma", "mb"]
+
+    label = "USD Add Maya Reference"
+    order = 1
+    icon = "code-fork"
+    color = "orange"
+
+    identifier_key = "ayon_identifier"
+
+    def load(self, context, name=None, namespace=None, options=None):
+
+        selection = list(iter_ufe_usd_selection())
+        assert len(selection) == 1, "Select only one PRIM please"
+        ufe_path = selection[0]
+        path = self.filepath_from_context(context)
+
+        import mayaUsdAddMayaReference
```

Comment:
Guess need to make sure mayaUsdPlugin needed to be loaded before all the usd loading actions?
```suggestion
         # Make sure we can load the plugin
        cmds.loadPlugin("mayaUsdPlugin", quiet=True)
        import mayaUsdAddMayaReference
```

Reply:
I'm not entirely sure - so better safe than sorry I suppose. :)